compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2385,7 +2385,22 @@ impl<'a> Parser<'a> {
             };
             let (pat, ty) = if is_name_required || this.is_named_param() {
                 debug!("parse_param_general parse_pat (is_name_required:{})", is_name_required);
-
+                if let token::Ident(symname, _) = this.token.kind
+                    && symname.as_str().eq("Sized")
```

Comment:
I think this is too specific of a suggestion. I would likely not merge this unless it can be generalized for all trait names, since it's very unique for people to be writing `impl 'lifetime Sized`.

Reply:
yes. you're right. It was a hack.
Now the code handles more specific cases for not only `'lifetime Size` as I removed `&& symname.as_str().eq("Sized")`